common/src/lib.rs

```diff
@@ -52,6 +61,19 @@ pub const GAS_VALUE_PREFIX: &[u8] = b"g::gas_tree";
 
 pub type ExitCode = i32;
 
+#[derive(Clone, Debug, Decode, Encode, PartialEq, TypeInfo)]
+pub struct Dispatch {
+    pub kind: DispatchKind,
+    pub message: Message,
+}
+
+impl Dispatch {
+    pub fn handle_reply(message: Message) -> Self {
```

Comment:
I'd rename this function to `new_handle_reply` to reflect its constructor-like nature.